sda-download/api/sda/sda.go

```diff
@@ -337,35 +327,39 @@ func Download(c *gin.Context) {
 		// set the user and server public keys that is send from htsget
 		log.Debugf("Got to setting the headers: %s", c.GetHeader("client-public-key"))
 		c.Header("Client-Public-Key", c.GetHeader("Client-Public-Key"))
-		c.Header("Server-Public-Key", c.GetHeader("Server-Public-Key"))
 	}
 
 	if c.Request.Method == http.MethodHead {
 
+		// Create headers for htsget, containing size of the crypt4gh header
+		reencKey := c.GetHeader("Client-Public-Key")
+		headerSize := bytes.NewReader(fileDetails.Header).Size()
+		// Size of the header in the archive
+		c.Header("Server-Additional-Bytes", fmt.Sprint(headerSize))
+		if reencKey != "" {
```

Comment:
Do we really care about the client key here? For now I think we can assume the same kinds of public keys, meaning the same header size, so "fixed" number.

Reply:
Both `server-additional-bytes` and `client-additional-bytes` are part of the spec from crypt4gh, although optional. I thought it would be good to have them set, so that we don't run into problems later, if someone would have different types of keys. But I might be overdoing things... @dbampalikis ?

Reply:
Sorry if this was unclear, I didn't mean we shouldn't send those but rather that the numbers we send won't depend on the public key used in the request, so there's no need to check for key here (they will likely depend on whatever we stick a data-edit-list in there, once we support that, so they will depend on whatever coordinates is requested but that can be dealt with later).

Possibly, one might move up the check for a valid key and bail out earlier if none is supplied if encrypted transfers are required, but I've already approved and am fine with merging this as is.

Reply:
to be picked up in #834 .